src/main/java/com/amazon/opendistroforelasticsearch/ad/model/DetectorValidationIssue.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package com.amazon.opendistroforelasticsearch.ad.model;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+
+import com.google.common.base.Objects;
+
+public class DetectorValidationIssue implements ToXContentObject, Writeable {
+    private static final String MESSAGE_FIELD = "message";
+    private static final String SUGGESTED_FIELD_NAME = "suggested_value";
+
+    private final ValidationAspect aspect;
+    private final DetectorValidationIssueType type;
+    private final String message;
+    private Map<String, String> subIssues;
+    private Object suggestion;
```

Comment:
Why `suggestion` is `Obeject` type ?

Reply:
For now I will leave it as place holder. Most likely it will be `String`, but it may change to other type if we need more complicated type for suggestion of model level validation. Will finalize it after model level validation is done.